### PR TITLE
migrate to lsp execution

### LINKED ISSUE #1974: Don't create the same planes multiple times
If you run the test [riddle_small.kcl](https://github.com/KittyCAD/modeling-app/blob/main/src/wasm-lib/tests/executor/inputs/riddle_small.kcl) and println the API calls, you'll see that the exact same `MakePlane` command is being sent twice. In the real riddle program, MakePlane is sent _many_ times. We shouldn't send multiple MakePlanes for the exact same plane, that's just causing unnecessary IO and potentially breaking batches.

### PR DESCRIPTION
**Pull request by jessfraz**
closes #1974

todo:

- [x] clear scene in between executions got broken in the move, figure out the right place for this
- [x] when you go from a long file to an empty file it should clear the scene
- [x] integrate the lsp notification for when the user changes their units
- [x] add a test for the diagnostics on an executor error in the rust code
- [x] add a test for the diagnostics on parser error in the rust code
- [x] fix default planes need to be cleared when we clear scene
- [x] small race condition on execute too fast on load and not getting values back
- [x] can we move the format extension in TextEditor.tsx to the lsp server since we already do that
- [x] update units should use the current file name for posperity
- [x] indent service to auto indent lines in a pipe on enter
- [x] fold service
- [x] figure out what on main is re-rendering on unit change? do we don't trigger it again
- [x] fix the command logs
- [x] empty file should be a valid ast
- [x] whitespace and comment only file should be valid ast
- [x] fix after clear cant start sketch
- [x] initialize planes in a batch so they dont flash
- [x] need to update kclErrors in kclManager when we throw a diagnostic, not being added to below pane
- [x] fix line highlight on draw lines
- [x] more efficient set isStreamReady don't recreate the lsp
- [x] codemirror has weird console errors on click and point
- [x] fix playwright tests
- [x] whenever we call `setCode` the lsp will execute since the text in the editor is changing make sure we dont have redundant calls
- [x] test making lots of changes in a large file 

I will come back for these in other prs:

- [ ] on unit change zoom to fit (once we have zoom to fit)
- [ ] use a worker for the lsp server
- [ ] do we stilll have to ignore updates when we are tabbing in a snippet?
- [ ] don't debounce parsing
- [ ] still could use some love on the speed side with canceling in progress executions
- [ ] delete any functions from kcl manager we no longer need, i think i got all these but can do another once over to be sure

after this, anytime you find yourself calling "re-execute" manually lmk because you may not need it, i cleaned a lot of them up

### CODE REVIEW

File: src/components/ModelingMachineProvider.tsx [hunk 1]
```diff
@@ -116,6 +118,24 @@ export const ModelingMachineProvider = ({
...
       actions: {
+        'disable lsp execution': async () => {
+          // Update the lsp server that we are in sketch mode, so we can turn off lsp execution.
+          for (const lspClient of lspClients) {
+            for (const plugin of lspClient.plugins) {
+              await plugin.updateCanExecute(false)
+            }
+          }
+          await kclManager.enterEditMode()
+          console.log('done with disabling lsp execution')
```
Comment (Irev-Dev) [inline src/components/ModelingMachineProvider.tsx:129]:
```suggestion
```

**Comment by jessfraz** [comment]
What's making me nervous is macOS failing and Ubuntu being super perfect

**Comment by jessfraz** [comment]
They all pass locally fine

**Review by Irev-Dev — APPROVED** [review]
Nice job!

I haven't check it out and played with it, but wanted to have a quick glance before I went to bed.

The macos playwright tests seem un happy, multiple failures, I know you re ran them and I just did again 🤞 

Was it intentional to have the cutter increase in width? that's needed for code folding yeah?

**Comment by jessfraz** [comment]
Yeah the gutter changes are intended

**Comment by jessfraz** [comment]
I’m gunna fuck w running macOS in a loop maybe there’s just some small races

**Comment by Irev-Dev** [comment]
I think 5 failing in CI and none failing locally is not going to reveal itself by running more times locally, but I guess worth a try,

I would start by looking at the playwright traces using https://trace.playwright.dev/

I can probably do this later.

**Comment by Irev-Dev** [comment]
I think https://github.com/KittyCAD/modeling-app/pull/2063 is likely the cause of all the image conflicts, as Frank also changed a lot of the snapshots, might need to rebase.

**Pull request closed without merging**